File: 11.盛最多水的容器.js
/*
 * @lc app=leetcode.cn id=11 lang=javascript
 *
 * [11] 盛最多水的容器
 */

// @lc code=start
/**
 * @param {number[]} height
 * @return {number}
 */
 var maxArea = function(height) {
    let left = 0;
    let right = height.length - 1;
    let ret = 0;
    while (left < right) {
        let h = height[left] < height[right] ? height[left++] : height[right--];
        ret = Math.max(ret, h * (right - left + 1));
    }
    return ret;
};
// @lc code=end

File: 11.盛最多水的容器.rs
/*
 * @lc app=leetcode.cn id=11 lang=rust
 *
 * [11] 盛最多水的容器
 */

// @lc code=start
impl Solution {
    pub fn max_area(height: Vec<i32>) -> i32 {
        let mut left = 0;
        let mut right = height.len() - 1;
        let mut ret = 0;
        while left < right {
            let h = if height[left] < height[right] { left += 1; height[left - 1] } else { right -= 1; height[right + 1] };
            ret = ret.max(h * (right - left + 1) as i32);
        }
        ret
    }
}
// @lc code=end

